src/recipes/login.tsx

```diff
@@ -0,0 +1,231 @@
+
+  onResendOtp = () => {
+    this.setState({ otpTimeout: false });
+    this.props.onResendOtp();
+  };
+
+  getOtpPage = () => {
+    const { loginUserValue, otpValue, onOtpChange, onSignIn } = this.props;
+
+    const { otpTimeout } = this.state;
+
+    return (
+      <>
+        <View style={styles.userInfoWrap}>
+          <Text style={styles.loginUserText}>{loginUserValue}</Text>
+          <Touchable
+            onPress={() => {
+              this.setState({ loginPage: LOGIN_PAGE.USER_PAGE });
+              onOtpChange("");
+            }}
+          >
+            <Text style={styles.textButton}>Edit</Text>
+          </Touchable>
+        </View>
+        <View style={styles.otpInputWrap}>
+          <Input
+            placeholder="Enter OTP"
+            onChange={onOtpChange}
+            value={otpValue}
+            style={styles.otpInput}
+          />
+          <View>
+            {otpTimeout && (
+              <Touchable onPress={this.onResendOtp}>
+                <Text style={styles.textButton}>Resend</Text>
+              </Touchable>
+            )}
+            {!otpTimeout && (
+              <Countdown
+                style={styles.countdownStyles}
+                onFinish={() => this.setState({ otpTimeout: true })}
+              />
+            )}
+          </View>
+        </View>
+        <Button style={{ marginTop: 50 }} onPress={onSignIn}>
+          Sign in
+        </Button>
+      </>
+    );
+  };
+
+  render() {
+    const { loginMethod, loginPage, sendingOTP } = this.state;
+    const {
+      onLoginUserChange,
+      loginUserValue,
+      countriesList,
+      onCountryChange,
+      selectedCountry
+    } = this.props;
+
+    return (
+      <View style={styles.container}>
+        <Text style={styles.loginHeader}>Glad to see you!</Text>
+        <Text style={styles.loginSubHeader}>Sign in to continue</Text>
+        <View style={styles.formContainer}>
+          {loginPage === LOGIN_PAGE.USER_PAGE && (
+            <>
+              <Controls
+                type="radio"
+                selected={loginMethod}
+                data={LOGIN_METHODS}
+                onChange={({ selected }) => {
+                  onLoginUserChange("");
+                  this.setState({ loginMethod: selected as number });
+                }}
+              />
+              <View style={styles.loginUserInput}>
+                {loginMethod === LOGIN_OPTIONS.EMAIL ? (
+                  <Input
+                    placeholder="Email Address"
+                    onChange={onLoginUserChange}
+                    value={loginUserValue}
+                    keyboardType="email-address"
+                  />
+                ) : (
+                  <>
+                    <View style={styles.countrySelect}>
+                      <Select
+                        options={countriesList}
+                        valueExtractor={item => item && item.country_code}
+                        rowLabelExtractor={item =>
```

Comment:
Under the assumption that item is not null.
`({name,country_code:countryCode}) => `${name} $(countryCode)` `

Reply:
The component takes value from the options prop. You already know whether it's null or not. unless you are creating options dynamically that might give the value null.